diff --git a/src/tasks-runner/forked-process-task-runner.js b/src/tasks-runner/forked-process-task-runner.js
index dcacef4a8b2e660a09fe5a528f79e3c311f04bee..589750ed681787e6cd32ce92a51f47eca2f54099 100644
--- a/src/tasks-runner/forked-process-task-runner.js
+++ b/src/tasks-runner/forked-process-task-runner.js
@@ -41,6 +41,9 @@ class ForkedProcessTaskRunner {
                     env: this.getEnvVariablesForProcess(),
                 });
                 this.processes.add(p);
+                // ensure the child process is killed when the parent exits
+                process.on('exit', () => p.kill());
+                process.on('SIGTERM', () => p.kill());
                 p.once('exit', (code, signal) => {
                     this.processes.delete(p);
                     if (code === null)